### ISSUE TITLE
Bogus error MD057 for inline `A|B`

### ISSUE THREAD

**Issue by nirs** [markdownlint/markdownlint#472]
## Description

Since 0.13.0 this document

```
This is | not a table
```

fails with:

```
test.md:1: MD055 Table row doesn't begin/end with pipes
test.md:2: MD057 Table has missing or invalid header separation (second row)
```

The first error does not happen in all cases, but it is a good hint - if
the raw does not begin with pipe it is not a able.

### Environment

mdl version 0.13.0

### Expected Behavior

Treat tables as they are treated by github/gilab parser. If the text is not displayed
as a table, the linter should not treat it as a table.

### Actual Behavior

Linter fails with bogus error about tables.

### Test

The following text shows that github does not treat the following test as a table. 

This is | not a table

**Comment by nirs** [comment]
Minimized original text showing this issue:

```
# Title

## Heading

- `status.conditions`:
    - `status`
        - "False" if reconciliation is progressing or has errors
            - `reason`: Progressing|Error
- `status.protectedPVCs`:
    - `pvcName`: Name of the PVC for which conditions are represented
```

Using this style:

```
all

rule 'MD007', :indent => 4
rule 'MD009', :br_spaces => 2
rule 'MD013', :ignore_code_blocks => true, :tables => false
rule 'MD024', :allow_different_nesting => true

exclude_rule 'MD040' # Fenced code blocks should have a language specified
exclude_rule 'MD041' # First line in file should be a top level header
```

With this we get only MD057:

```
$ mdl --style style.rb test.md 
test.md:9: MD057 Table has missing or invalid header separation (second row)
```

**Comment by jaymzh** [comment]
Yup, I noticed that as well. Mentioned it in the original PR here: https://github.com/markdownlint/markdownlint/pull/464#issuecomment-1744030620

**Comment by nbehrnd** [comment]
@nirs For conversions of .md into either .org, or .html with pandoc (version 3.1.9), I noticed a file like

```markdown
In a simple text, an escape with \| equally displays the pipe character.
```

provides a bypass for which a check with mdl (packaged version 0.13.0) does not yields a warning.

**Comment by nirs** [comment]
> provides a bypass for which a check with mdl (packaged version 0.13.0) does not yields a warning.

Thanks, we use this workaround now, see
https://github.com/RamenDR/ramen/pull/1077/commits/2b25c1405d1726ea2080d3844bd5bbac8ad67af2

**Comment by nbehrnd** [comment]
@nirs Your approach identified to change the source code of `markdownlint` does not appear yet in the commit history of the main branch since tag `0.13.0`, nor in the list of push requests pending/closed since October 1st.  So it is a detail for a future version, ok.

**Comment by jaymzh** [comment]
The offending PR has been reverted.
